### PR TITLE
SDL2: 2.30.6 -> 2.30.9

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Yay! A mostly harmless refactor found a bug in the test packages!

File: pkgs/by-name/sd/SDL2/package.nix [hunk 1]
```diff
@@ -160,15 +159,19 @@ stdenv.mkDerivation (finalAttrs: {
...
-  enableParallelBuilding = true;
-
   configureFlags =
     [ "--disable-oss" ]
     ++ lib.optional (!x11Support) "--without-x"
     ++ lib.optional alsaSupport "--with-alsa-prefix=${alsa-lib.out}/lib"
     ++ lib.optional stdenv.hostPlatform.isWindows "--disable-video-opengles"
     ++ lib.optional stdenv.isDarwin "--disable-sdltest";
 
+  dontDisableStatic = if withStatic then 1 else 0;
```
Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix]: Isn't nix converting this truthy, so we can do

```suggestion
  dontDisableStatic = withStatic;
```

File: pkgs/by-name/sd/SDL2/package.nix [hunk 2]
```diff
@@ -74,17 +74,18 @@ stdenv.mkDerivation (finalAttrs: {
...
+
   outputBin = "dev"; # sdl-config
 
   patches = [
-    # `sdl2-config --cflags` from Nixpkgs returns include path to just SDL2.
-    # On a normal distro this is enough for includes from all SDL2* packages to work,
-    # but on NixOS they're spread across different paths.
+    # `sdl2-config --cflags` from Nixpkgs returns include path to just SDL2. On
+    # a normal distro this is enough for includes from all SDL2* packages to
+    # work, but on NixOS they're spread across different paths.
```
Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix:131]: For me, this is now harder to read...

File: pkgs/by-name/sd/SDL2/package.nix [hunk 3]
```diff
@@ -166,29 +166,28 @@ stdenv.mkDerivation (finalAttrs: {
...
-    fi
-    moveToOutput bin/sdl2-config "$dev"
-  '';
+  postInstall =
+    (if withStatic then
+      ''rm $out/lib/*.la''
+     else
+       ''rm $out/lib/*.a'')
+    + "\n" +
+    ''moveToOutput bin/sdl2-config "$dev"'';
```
Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix]:
```suggestion
  postInstall =
    (if withStatic then
      ''
        rm $out/lib/*.la
      ''
     else
       ''
         rm $out/lib/*.a
       '')
    + ''
      moveToOutput bin/sdl2-config "$dev"
    '';
```

File: pkgs/by-name/sd/SDL2/package.nix [hunk 4]
```diff
@@ -189,19 +189,18 @@ stdenv.mkDerivation (finalAttrs: {
     + "\n" +
     ''moveToOutput bin/sdl2-config "$dev"'';
 
-  # SDL is weird in that instead of just dynamically linking with
```
Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix:192]: I wouldn't mind if the big comment block was not changed multiple times...

File: pkgs/by-name/sd/SDL2/package.nix [hunk 5]
```diff
@@ -63,6 +63,41 @@
 # SDL expression too
 let
   inherit (darwin.apple_sdk.frameworks) AudioUnit Cocoa CoreAudio CoreServices ForceFeedback OpenGL;
+  dlopenPropagatedBuildInputs =
+    [ ]
```
Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix:75]:
```suggestion
```

File: pkgs/by-name/sd/SDL2/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix:1]: > SDL2: lib.optional -> lib.optionals

I mean it doesn't really matter in the end 😓

Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix:1]: > SDL2: new parameter enableSdltest

Whats the usecase for it? another platform like musl?

Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix:1]: > SDL2: nixfmt-rfc-style

Can we fix up those in the respective commits?

File: pkgs/by-name/sd/SDL2/package.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/by-name/sd/SDL2/package.nix]: Outdated - done.

File: pkgs/by-name/sd/SDL2/package.nix (hunk 5 above)
Comment (AndersonTorres) [reply pkgs/by-name/sd/SDL2/package.nix]: Done

File: pkgs/by-name/sd/SDL2/package.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/by-name/sd/SDL2/package.nix]: Formatter undid this.

File: pkgs/by-name/sd/SDL2/package.nix [hunk 6]
```diff
@@ -166,7 +166,7 @@ stdenv.mkDerivation (finalAttrs: {
     ++ lib.optional stdenv.hostPlatform.isWindows "--disable-video-opengles"
     ++ lib.optional stdenv.hostPlatform.isDarwin "--disable-sdltest";
 
-  dontDisableStatic = if withStatic then 1 else 0;
+  dontDisableStatic = withStatic;
```
Comment (SuperSandro2000) [inline pkgs/by-name/sd/SDL2/package.nix]: This seems to be unused now.

```suggestion
```

Comment (AndersonTorres) [reply pkgs/by-name/sd/SDL2/package.nix]: This parameter defines which files will be removed  in `postInstall`.

**Comment by AndersonTorres** [comment]
Mental note: study the new Darwin SDK.

**Comment by trofi** [comment]
`2.30.9` is out: https://github.com/libsdl-org/SDL/releases/tag/release-2.30.9

File: pkgs/by-name/sd/SDL2/package.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/by-name/sd/SDL2/package.nix:193]: Solved.
But ugly.

**Comment by AndersonTorres** [comment]
As far as I can see, the bugs are in the other packages. Let's wait.

**Comment by AndersonTorres** [comment]
Finally it is working!

**Pull request closed without merging**